### PR TITLE
Add method to bind to any port

### PR DESCRIPTION
**Pull request by BourgondAries**

### CODE REVIEW

File: src/net/socket.rs [hunk 1]
```diff
@@ -224,6 +239,11 @@ mod tests {
     use std::thread;
     use std::time::{Duration, Instant};
 
+    #[test]
+    fn binding_to_any() {
+        Socket::bind_any().unwrap();
```
Comment (TimonPost) [inline src/net/socket.rs]: I would prefer to `assert!(Socket::bind_any().is_ok())`, but that is up to you.

File: src/net/socket.rs [hunk 2]
```diff
@@ -38,6 +38,14 @@ impl Socket {
         Socket::bind_with_config(addresses, Config::default())
     }
 
+    /// Bind to any local port on the system, if available
+    pub fn bind_any() -> Result<(Self, Sender<Packet>, Receiver<SocketEvent>)> {
```
Comment (jstnlef) [inline src/net/socket.rs]: Probably want a version of this which takes a Config

Comment (BourgondAries) [reply src/net/socket.rs]: Done

File: src/net/socket.rs [hunk 3]
```diff
@@ -38,6 +38,24 @@ impl Socket {
...
+        let loopback = Ipv4Addr::new(127, 0, 0, 1);
```
Comment (jstnlef) [inline src/net/socket.rs]: Now this should call `bind_any_with_config(Config::default())`

Comment (BourgondAries) [reply src/net/socket.rs]: Done

**Pull request merged**